main/buildgen/src/mill/main/buildgen/BuildGenUtil.scala

```diff
@@ -0,0 +1,363 @@
+package mill.main.buildgen
+
+import mill.main.buildgen.BuildObject.Companions
+import mill.main.client.CodeGenConstants.{
+  buildFileExtensions,
+  nestedBuildFileNames,
+  rootBuildFileNames,
+  rootModuleAlias
+}
+import mill.main.client.OutFiles
+import mill.runner.FileImportGraph.backtickWrap
+
+@mill.api.internal
+object BuildGenUtil {
+
+  type BuildNode = Node[BuildObject]
+  type BuildPackage = String
+  type BuildTree = Tree[BuildNode]
+
+  type Gav = (String, String, String)
+  type InterpIvy = String
+
+  type BomIvyDeps = IterableOnce[String] // interpolated or named
+  type IvyDeps = IterableOnce[String] // inmergeterpolated or named
+  type ModuleDeps = IterableOnce[BuildPackage]
```

Comment:
I think most of these aliases aren't pulling their weight in making the code easier to read v.s. the indirection they impose.

Let's get rid of them and use the underlying types directly